### ISSUE TITLE
Duplicate job was pushed ( v6.0.13 )

### ISSUE THREAD

**Issue by metapraveen** [mhenrixon/sidekiq-unique-jobs#414]
**Describe the bug**
We have 2 instances running sidekiq, I used this gem to avoid duplication but got duplication. 
Gem version 6.0.13

**Expected behavior**
Jobs pushed should have been unique

**Current behavior**
I see 2 similar jobs getting pushed

**Worker class**

```ruby
class MyWorket
  include Sidekiq::Worker
  sidekiq_options retry: false, lock: :until_executed, on_conflict: :log, log_duplicate_payload: true
  
  def perform(); end
end
```

**Additional context**
I would like to debug the gem and see if I can get to the issue, but I may need a starting point to look in the right direction.

**Comment by mhenrixon** [comment]
@metapraveen that is a new issue. Version 6 was always pretty bad at keeping locks around for longer than necessary but I haven't had any complaints about duplicate locks.

Are you sure that you are not just seeing a log about duplication because of the `on_conflict`? If you add a sleep for a few seconds will there really be two jobs executing at the same time?

**Comment by metapraveen** [comment]
My jobs supposed to make x number of HTTP calls, but it makes 2x in production which means the job is duplicated. 
I tried to reproduce this in local by adding a lot of redis latency and pushing around 25 jobs at a time, but this gem seems to handle it perfectly and only one job is executed. 
Wish I could replicate this in local easily :(

**Comment by kolo** [comment]
I spent some time trying to debug this issue and came to the following conclusion. The bug happens when there is a conflict and log strategy is called. The strategy uses `log_info` method which returns `true` value. This value becomes return value of the [`BaseLock#lock`](https://github.com/mhenrixon/sidekiq-unique-jobs/blob/v6.0.13/lib/sidekiq_unique_jobs/lock/base_lock.rb#L32) method. `Client::Middleware#call` [interprets this value as if locking was successful and yields next middleware in the chain](https://github.com/mhenrixon/sidekiq-unique-jobs/blob/v6.0.13/lib/sidekiq_unique_jobs/client/middleware.rb#L27). This results it queuing of a duplicate job.

I believe this issue has been solved in the `master` branch because all logging methods in it return `nil` explicitly.

**Comment by mhenrixon** [comment]
Ah, I'll cut a new release @kolo ! Thanks a bunch for taking the time to both investigate and then present your findings.

**Comment by mhenrixon** [comment]
Fixed in `v6.0.14`
